### PR TITLE
docs: add Roadmap

### PR DESCRIPTION
**Pull request by tinytb**
EngRes: please feel free to push edits directly to this branch; this is a collaboratively-edited document.

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+## Table of Contents <!-- omit in toc -->
+
+- [🛣️ Milestones](#️-milestones)
+  - [2022](#2022)
+    - [Mid Q4 (November)](#mid-q4-november)
+    - [Late Q4 (December)](#late-q4-december)
+  - [2023](#2023)
+    - [Early Q1 (January)](#early-q1-january)
+    - [Late Q1 (March)](#late-q1-march)
+    - [Q2](#q2)
```
Comment (BigLep) [inline ROADMAP.md:22]: We should include the milestones themselves as subbullets.  Right now someone needs to read to see this.

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Roadmap <!-- omit in toc -->
+
+This project is creating a focused, modular, opinionated, JavaScript-native implementation of [IPFS](https://ipfs.tech/) (the Interplanetary File System).
+
+Our goal is a high-quality implementation of the IPFS protocol in TypeScript/JavaScript. It shall run in web browsers, in service workers, in browser extensions, in Node.js, and virtually everywhere else JS runs.
+
+```
+Date: 2022-10-26
+Status: Draft
+Notes: Internal stakeholders have not yet aligned on this roadmap. Please add any feedback or questions in the PR.
```
Comment (BigLep) [inline ROADMAP.md:10]:
```suggestion
Notes: Maintainers have aligned on this roadmap. Please add any feedback or questions in:
https://github.com/ipfs/pomegranate/issues/5
```

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    - [Early Q1 (January)](#early-q1-january)
+    - [Late Q1 (March)](#late-q1-march)
+    - [Q2](#q2)
+
+## 🛣️ Milestones
+### 2022
+
+#### Mid Q4 (November)
+
+Communicate the IPFS-in-JS state.
```
Comment (BigLep) [inline ROADMAP.md:29]: I think it would be ideal to make these subheadings so they can be linked to.

File: ROADMAP.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+**Problem to solve:** Currently, very few know the direction for IPFS-in-JS and how they can best help. This affects project resourcing, recruiting, and IPFS adoption in general.
+
+**Done state:**
+- Present and share the IPFS Camp 2022 presentation.
+- Write and publish a blog post.
+- Hold a community vote and communication about a name for the new IPFS-in-JS implementation.
```
Comment (BigLep) [inline ROADMAP.md:36]: Ideally each of the milestones has a tracking issue that someone can follow.

File: ROADMAP.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+- Project scope, milestones, success criteria, and communication channels are established.
+- Community can follow along and contribute to "Pomegranate".
+
+#### Late Q1 (March)
+
+"Pomegranate" "v1" released.
+
+- Users can add and get files.
+- Packaging, publishing, testing, CI/CD, etc. are all set up.
```
Comment (BigLep) [inline ROADMAP.md:62]:
```suggestion
- Packaging, publishing, testing, CI/CD, etc. are all set up.

Tracking issue: https://github.com/ipfs/pomegranate/issues/2
```

**Review by BigLep — APPROVED** [review]
@tinytb : I'm going to approve and merge this.  Please go ahead and incorporate comments after the fact.  I also created this accompanying issue: https://github.com/ipfs/pomegranate/issues/5

File: ROADMAP.md (hunk 1 above)
Comment (tinytb) [reply ROADMAP.md:23]: https://github.com/ipfs/pomegranate/pull/8

**Comment by tinytb** [comment]
Follow-up PR:
- https://github.com/ipfs/pomegranate/pull/8

**Pull request merged**